newIDE/app/src/VariablesList/VariablesList.js

```diff
@@ -94,11 +94,16 @@ export type HistoryHandler = {|
 type Props = {|
   variablesContainer: gdVariablesContainer,
   inheritedVariablesContainer?: gdVariablesContainer,
+  /** Callback executed at mount to compute suggestions. */
   onComputeAllVariableNames?: () => Array<string>,
+  /** To specify if history should be handled by parent. */
   historyHandler?: HistoryHandler,
   emptyPlaceholderTitle?: React.Node,
   emptyPlaceholderDescription?: React.Node,
   helpPagePath?: ?string,
+  /** If set to false, it will commit changes to variables on each input change. It can be expensive, but useful when VariablesList can be unmounted at any time. */
```

Comment:
I was thinking we could even change the SemiControlledTextField to throttle the commits: commit X milliseconds after the last change.
So you get "real time updates" but at the same time not paying for a rendering at each keystroke (only ~500ms after the last change).

Anyway, thanks for the comment, it will work for now!

Reply:
Oh yes, that could do the trick if there are performance issues!